src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -93,17 +93,30 @@ private static ImmutableArray<Document> FilterDocumentsByClientName(ImmutableArr
             if (documents.Length > 1)
             {
                 // We have more than one document; try to find the one that matches the right context
-                if (documentIdentifier is VSTextDocumentIdentifier vsDocumentIdentifier)
+                if (documentIdentifier is VSTextDocumentIdentifier vsDocumentIdentifier && vsDocumentIdentifier.ProjectContext != null)
                 {
-                    if (vsDocumentIdentifier.ProjectContext != null)
+                    var projectId = ProtocolConversions.ProjectContextToProjectId(vsDocumentIdentifier.ProjectContext);
+                    var matchingDocument = documents.FirstOrDefault(d => d.Project.Id == projectId);
+
+                    if (matchingDocument != null)
                     {
-                        var projectId = ProtocolConversions.ProjectContextToProjectId(vsDocumentIdentifier.ProjectContext);
-                        var matchingDocument = documents.FirstOrDefault(d => d.Project.Id == projectId);
+                        return matchingDocument;
+                    }
+                }
+                else
+                {
+                    // We were not passed a project context.  This can happen when the LSP powered NavBar is not enabled.
+                    // This branch should be removed when we're using the LSP based navbar in all scenarios.
+
+                    // Lookup the active document and determine if any of the documents from the request URI match.
+                    var solution = documents.First().Project.Solution;
+                    var service = solution.Workspace.Services.GetRequiredService<IDocumentTrackingService>();
```

Comment:
instead of IDocumentTrackingService just use: https://github.com/dotnet/roslyn/blob/eb0255504d2a4064cdcd6b4aaf0a4d0c0140ea97/src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs#L262

Reply:
Ahhh of course!  Will switch to that